### PR TITLE
[Task]: Snowflake Driver Support

### PR DESCRIPTION
**Pull request by shossain-mizzou**
- Snowflake JDBC driver support for all DDL
- Demo data installation provided for ACT ontology

### CODE REVIEW

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/snowflake/create_i2b2metadata_tables.sql
Comment (jklann) [inline edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/snowflake/create_i2b2metadata_tables.sql:1]: This is duplicative of the copy in act/scripts/create_snowflake_i2b2_metadata. Not sure why both are here. Please follow the convention of the other db platforms.

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/table_access_insert_data_oracle.sql
Comment (jklann) [inline edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/table_access_insert_data_oracle.sql:1]: I see the problem you're trying to solve here - you really need an "if not snowflake" switch. Although if it does need to be done this way, would be better to follow the convention of scripts/[dbtype]/insert_table_access.sql

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/data_build.xml [hunk 1]
```diff
@@ -133,10 +133,10 @@ END;
 		</unzip>
 
 		<sql driver="${db.driver}" url="${db.url}" userid="${db.username}" password="${db.password}" classpathref="classpath" onerror="continue">
-			<transaction src="./${db.project}/scripts/schemes_insert_data.sql" />
+			<transaction src="./${db.project}/scripts/schemes_insert_data_${db.type}.sql" />
```
Comment (jklann) [inline edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/data_build.xml:136]: again, better to follow convention scripts/[dbtype]/scriptname.sql

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/scripts/procedures/.DS_Store
Comment (jklann) [inline edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/scripts/procedures/.DS_Store:1]: Please remove these from your commit, it messes up the repo.

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Pmdata/db.properties
Comment (jklann) [inline edu.harvard.i2b2.data/Release_1-7/NewInstall/Pmdata/db.properties:1]: same comment here and in others, don't change default db.

**Review by jklann — CHANGES_REQUESTED** [review]
Actual code looks fine, naming conventions and defaults need to be changed.

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Crcdata/db.properties
[the remark this answers is not in the document]
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Crcdata/db.properties:1]: Oracle is now default configuration

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Hivedata/db.properties
[the remark this answers is not in the document]
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Hivedata/db.properties:1]: Oracle is now default configuration

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/schemes_insert_data_sqlserver.sql
[the remark this answers is not in the document]
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/schemes_insert_data_sqlserver.sql:1]: I reverted back the file modification and added condition
sql```
<condition property="not-snowflake">
		<not>
			<equals arg1="${db.type}" arg2="snowflake" />
		</not>
	</condition>

	<target name="db_metadata_insert_schemes_data" if="not-snowflake">
		<sql driver="${db.driver}" url="${db.url}" userid="${db.username}" password="${db.password}" classpathref="classpath" onerror="continue">
			<transaction src="./${db.project}/scripts/schemes_insert_data.sql" />
		</sql>
	</target>

	<target name="db_metadata_insert_table_access_data" if="not-snowflake">
		<sql driver="${db.driver}" url="${db.url}" userid="${db.username}" password="${db.password}" classpathref="classpath" onerror="continue">
			<transaction src="./${db.project}/scripts/table_access_insert_data.sql" />
		</sql>
	</target>
>
<antcall target="db_metadata_insert_schemes_data"/> 
<antcall target="db_metadata_insert_table_access_data"/>
```

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/db.properties
[the remark this answers is not in the document]
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/db.properties:1]: Oracle is now default configuration

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/snowflake/create_i2b2metadata_tables.sql
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/snowflake/create_i2b2metadata_tables.sql:1]: Fixed

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/table_access_insert_data_oracle.sql
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/act/scripts/table_access_insert_data_oracle.sql:1]: Fixed

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/data_build.xml (hunk 1 above)
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/data_build.xml]: Fixed

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/scripts/procedures/.DS_Store
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Metadata/scripts/procedures/.DS_Store:1]: Removed

File: edu.harvard.i2b2.data/Release_1-7/NewInstall/Pmdata/db.properties
Comment (shossain-mizzou) [reply edu.harvard.i2b2.data/Release_1-7/NewInstall/Pmdata/db.properties:1]: Fixed

**Comment by shossain-mizzou** [comment]
@jklann, I have addressed all the changes. Would you please review again?

**Review by jklann — APPROVED** [review]
Looks good now, thanks for making those changes.

**Pull request merged**